src/main/java/org/broadinstitute/hellbender/tools/gvs/common/WeightedSplitIntervals.java

```diff
@@ -0,0 +1,219 @@
+
+        // calculate total weights and target weight per shard
+        float totalWeight = 0;
+        for (Interval i : intervalsWithWeights) {
+            totalWeight += ((WeightedInterval) i).getWeight();
+        }
+        float targetWeightPerScatter = totalWeight / (float) scatterCount;
+
+        System.out.println("Total Weight:"+totalWeight + " scatterCount: " + scatterCount + " target:" + targetWeightPerScatter);
+
+        final int maxNumberOfPlaces = Math.max((int)Math.floor(Math.log10(scatterCount-1))+1, numDigits);
+        final String formatString = "%0" + maxNumberOfPlaces + "d";
+
+        int scatterPiece = 0;
+        float cumulativeWeight = 0;
+        IntervalList currentList = new IntervalList(sequenceDictionary);
+        String lastContig = null;
+
+        PushbackIterator<Interval> iter = new PushbackIterator<>(intervalsWithWeights.iterator());
+        while (iter.hasNext()) {
+            WeightedInterval wi = (WeightedInterval) iter.next();
+
+            // if we're not mixing contigs, but we switched contigs, emit the list
+            if (dontMixContigs && lastContig != null && !lastContig.equals(wi.getContig()) ) {
+                // write out the current list (uniqued and sorted) and start a new one
+                currentList.uniqued().sorted().write(new File(outputDir, prefix + String.format(formatString, scatterPiece++) + extension));
+                currentList = new IntervalList(sequenceDictionary);
+                cumulativeWeight = 0;
+            }
+
+            // in either case, set the last seen contig
+            lastContig = wi.getContig();
+
+            // if the interval fits completely, just add it
+            if (cumulativeWeight + wi.getWeight() <= targetWeightPerScatter ) {
+                cumulativeWeight += wi.getWeight();
+                currentList.add(wi);
+
+            // if it would push us over the edge
+            } else {
+                // add a piece of it
+                float remainingSpace = targetWeightPerScatter - cumulativeWeight;
+
+                // how many bases can we take?
+                int basesToTake = (int) Math.floor(remainingSpace /  wi.getWeightPerBase());
+
+                // split and add the first part into this list
+                WeightedInterval[] pair = wi.split(basesToTake);
+                currentList.add(pair[0]);
+
+                // push the remainder back onto the iterator
+                iter.pushback(pair[1]);
+
+                // add uniqued, sorted output list and reset
+                currentList.uniqued().sorted().write(new File(outputDir, prefix + String.format(formatString, scatterPiece++) + extension));
+                currentList = new IntervalList(sequenceDictionary);
+                cumulativeWeight = 0;
+            }
+        }
+        // write the final list
+        currentList.uniqued().sorted().write(new File(outputDir, prefix + String.format(formatString, scatterPiece++) + extension));
```

Comment:
Lets make this file name construction into a function since it's repeated.  We could move it into the newly created `OutputScatterArgumentCollection` if we went that route.

Reply:
done, but this doesn't feel a whole lot cleaner 🤷